classes/category/Category.php

```diff
@@ -14,8 +14,13 @@
 
 namespace PKP\category;
 
+use PKP\security\Role;
+
 class Category extends \PKP\core\DataObject
 {
+    public const ASSIGNABLE_ROLES = [Role::ROLE_ID_MANAGER, Role::ROLE_ID_SUB_EDITOR, Role::ROLE_ID_ASSISTANT];
```

Comment:
I didn't review the visibility modifiers of anything in this PR, but in general I prefer to use private > protected > public, to decrease the surface of the public API and decrease breaking changes in the future. And if turning things into `private` makes it tough to customize/re-use stuff, then we probably have structural/organization issues (e.g. classes doing too much things).

Does everything that's marked as public should be public? If not, then it's better to hide the implementation.